src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/MonadBlueprint.hs

```diff
@@ -69,7 +71,7 @@ type ClosedPoly i a = forall x . Algebra a x => (i -> x) -> x
 -- NOTE: the property above is correct by construction for each function of a
 -- suitable type, you don't have to check it yourself.
 
-class Monad m => MonadBlueprint i a m | m -> i, m -> a where
+class Monad m => MonadBlueprint i a n m | m -> i, m -> a, m -> n where
```

Comment:
With separate `Circuit` type in place, we can work with circuits of arbitrary size and not have `n`  as a class parameter

Reply:
You don't _need_ a separate type but `n` should equal `0` in the state updates. Outputs should be passed in the return variable, not the state variable.